src/objects/js/components/admin/permissions/permission-form.js

```diff
@@ -1,17 +1,51 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 
 import { CheckboxInput, TextInput, SelectInput } from "../../forms/inputs";
 import { versionAuthFields } from "./auth-fields";
 
 
-const PermissionForm = ({objectFields, dataFieldChoices, tokenChoices, objecttypeChoices, modeChoices, formData}) => {
+const PermissionForm = ({objectFields, tokenChoices, objecttypeChoices, modeChoices, formData}) => {
     const {values, errors} = formData;
     const [mode, setMode]  = useState(values["mode"]);
     const [useFields, setUseFields] = useState(values["use_fields"]);
     const [objectType, setObjectType] = useState(values["object_type"]);
-
-    const [fields, setFields] = useState( JSON.parse(values["fields"]) || {});
-
+    if (!values["fields"]) {
+        values["fields"] = "{}"
+    }
+
+    const [fields, setFields] = useState( JSON.parse(values["fields"]) || {} )
+    const [dataFieldChoices, setDataFieldChoices] = useState(dataFieldChoices);
```

Comment:
```suggestion
    const [dataFieldChoices, setDataFieldChoices] = useState({});
```
Initial state should be different from the current state